gcloud/monitoring/test_client.py

```diff
@@ -157,6 +157,127 @@ def test_metric_descriptor_factory(self):
+        LABELS = {
+            'instance_id': 'my-instance-id',
+            'zone': 'us-central1-f'
+        }
+
+        client = self._makeOne(project=PROJECT, credentials=_Credentials())
+        client.connection = _Connection()   # For safety's sake.
+        resource = client.resource(TYPE, LABELS)
+        self.assertEqual(resource.type, TYPE)
+        self.assertEqual(resource.labels, LABELS)
+
+    def test_timeseries_factory(self):
+        import datetime
+        from gcloud._testing import _Monkey
+        import gcloud.monitoring.client
+        METRIC_TYPE = 'custom.googleapis.com/my_metric'
+        MY_CUMULATIVE_METRIC = 'custom.googleapis.com/my_cumlative_metric'
+        METRIC_LABELS = {
+            'instance_name': 'my-instance'
+        }
+
+        RESOURCE_TYPE = 'https://cloud.google.com/monitoring/api/resources'
+        RESOURCE_LABELS = {
+            'project_id': 'my-project-id',
+            'instance_id': 'my-instance-id',
+            'zone': 'us-central1-f'
+        }
+
+        VALUE = 42
+        VALUE2 = 43
+        VALUE3 = 44
+
+        TIME1 = datetime.datetime.utcnow()
+        TIME2 = datetime.datetime.utcnow()
+        TIME3 = datetime.datetime.utcnow()
+        RESET_TIME = datetime.datetime.utcnow()
+        TIME4 = datetime.datetime.utcnow()
+
+        # Construct a time series assuming a gauge metric without a start time
+        client = self._makeOne(project=PROJECT, credentials=_Credentials())
+        client.connection = _Connection()   # For safety's sake.
+        metric = client.metric(METRIC_TYPE, METRIC_LABELS)
+        resource = client.resource(RESOURCE_TYPE, RESOURCE_LABELS)
+        timeseries = client.time_series(metric, resource, VALUE,
+                                        end_time=TIME1)
+
+        self.assertEqual(timeseries.metric, metric)
+        self.assertEqual(timeseries.resource, resource)
+        self.assertEqual(len(timeseries.points), 1)
+        self.assertEqual(timeseries.points[0].value, VALUE)
+        self.assertIsNone(timeseries.points[0].start_time)
+        self.assertEqual(timeseries.points[0].end_time, TIME1)
+
+        # Construct a time series assuming a guage metric with an explicit
+        # start time
+        timeseries_with_start = client.time_series(metric,
+                                                   resource,
+                                                   VALUE,
+                                                   start_time=TIME1,
+                                                   end_time=TIME2)
+        self.assertEqual(timeseries_with_start.points[0].start_time, TIME1)
+        self.assertEqual(timeseries_with_start.points[0].end_time, TIME2)
+
+        # Construct a time series assuming a cumulative metric.
+        # CUMULATIVE metrics use the same start time repeatedly, until
+        # at some point they are reset with a new start time. We call
+        # the start time RESET_TIME to indicate this restart.
```

Comment:
I see what you are saying, but I wasn't thinking that illustrating a reset would be worth the trouble. In Monarch's terminology, it's called the reset time even if it's the first. I think two examples will suffice to illustrate the semantics: 1) start_time=RESET_TIME, end_time=TIME1; 2) start_time=RESET_TIME, end_time=TIME2.